executor/adapter.go

```diff
@@ -774,6 +774,11 @@ func (a *ExecStmt) handlePessimisticDML(ctx context.Context, e Executor) error {
 			return nil
 		}
 		e, err = a.handlePessimisticLockError(ctx, err)
+		// If it's not a retryable error, rollback current transaction instead of rolling back current statement like
+		// in normal transactions, because we cannot locate and rollback the statement that leads to the lock error.
+		if sctx.GetSessionVars().ConstraintCheckInPlacePessimistic {
+			sctx.GetSessionVars().SetInTxn(false)
```

Comment:
Should you put  these code in the code block `if err != nil {}` bellow which breaks the loop. The current process may lead the transaction status is `NotInTxn` but the retry succeeded.

Reply:
Right